File: ontrack-extension-ldap/src/main/java/net/nemerosa/ontrack/extension/ldap/LDAPMappingResourceDecorator.java
package net.nemerosa.ontrack.extension.ldap;

import net.nemerosa.ontrack.ui.resource.AbstractResourceDecorator;
import net.nemerosa.ontrack.ui.resource.Link;
import net.nemerosa.ontrack.ui.resource.ResourceContext;
import org.springframework.stereotype.Component;
import org.springframework.web.servlet.mvc.method.annotation.MvcUriComponentsBuilder;

import java.util.List;

@Component
public class LDAPMappingResourceDecorator extends AbstractResourceDecorator<LDAPMapping> {

    public LDAPMappingResourceDecorator() {
        super(LDAPMapping.class);
    }

    @Override
    public List<Link> links(LDAPMapping resource, ResourceContext resourceContext) {
        return resourceContext.links()
//                // Update
//                .link(Link.UPDATE, MvcUriComponentsBuilder.on(LDAPController.class).getMappingUpdateForm(resource.getId()))
//                        // Delete
//                .link(Link.DELETE, MvcUriComponentsBuilder.on(LDAPController.class).deleteMapping(resource.getId()))
                        // OK
                .build();
    }
}

File: ontrack-extension-ldap/src/main/java/net/nemerosa/ontrack/extension/ldap/LDAPMappingResourceDecorator.kt
package net.nemerosa.ontrack.extension.ldap

import net.nemerosa.ontrack.ui.resource.AbstractLinkResourceDecorator
import net.nemerosa.ontrack.ui.resource.Link
import net.nemerosa.ontrack.ui.resource.LinkDefinition
import net.nemerosa.ontrack.ui.resource.linkTo
import org.springframework.stereotype.Component
import org.springframework.web.servlet.mvc.method.annotation.MvcUriComponentsBuilder.on

@Component
class LDAPMappingResourceDecorator : AbstractLinkResourceDecorator<LDAPMapping>(LDAPMapping::class.java) {

    override fun getLinkDefinitions(): Iterable<LinkDefinition<LDAPMapping>> = listOf(
            Link.UPDATE linkTo { mapping -> on(LDAPController::class.java).getMappingUpdateForm(mapping.id) },
            Link.DELETE linkTo { mapping -> on(LDAPController::class.java).deleteMapping(mapping.id) }
    )

}